src/App.tsx

```diff
@@ -8,53 +9,151 @@
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { getUserPosts } from './api/ClientFunctions';
+import { Post } from './types/Post';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  // #region states
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const [currUser, setCurrUser] = useState<User | null>(null);
+  const [isDdActive, setIsDdActive] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [loadingPostsError, setLoadingPostsError] = useState(false);
+  const [arePostsLoading, setArePostsLoading] = useState(false);
+  const [isNoPosts, setIsNoPosts] = useState(false);
+  const [openedPost, setOpenedPost] = useState<Post | null>(null);
+  const [prevUser, setPrevUser] = useState<User | null>(null);
+  const [doesFormExist, setDoesFormExist] = useState(false);
 
-              <Loader />
+  // #endregion
+  // #region handlers
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const sectionOnClickHandler = () => {
+    if (isDdActive) {
+      setIsDdActive(false);
+    }
+  };
+
+  // #endregion
+  // #region useEffects
+
+  useEffect(() => {
+    if (currUser) {
+      const { id } = currUser;
+
+      setArePostsLoading(true);
+
+      getUserPosts(id)
+        .then(posts => {
+          setUserPosts(posts);
+
+          if (loadingPostsError) {
+            setLoadingPostsError(false);
+          }
+        })
+        .catch(() => setLoadingPostsError(true))
+        .finally(() => setArePostsLoading(false));
+    }
+  }, [currUser]);
+
+  useEffect(() => {
+    if (userPosts.length === 0 && currUser) {
+      setIsNoPosts(true);
+    } else if (userPosts.length > 0 && isNoPosts) {
+      setIsNoPosts(false);
+    }
+  }, [currUser, isNoPosts, userPosts]);
+
+  useEffect(() => {
+    if (prevUser?.id !== currUser?.id) {
+      setOpenedPost(null);
+    }
+  }, [prevUser, currUser]);
+
+  useEffect(() => {
+    if (doesFormExist) {
+      setDoesFormExist(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [openedPost]);
```

Comment:
The `useEffect` dependency array includes `openedPost`. If the effect is intended to run only when `openedPost` changes, ensure that the logic inside the effect is necessary for this dependency. Otherwise, it might lead to unnecessary re-renders.